main.tf

```diff
@@ -307,15 +314,12 @@ resource "aws_s3_bucket" "origin" {
 }
 
 resource "aws_s3_bucket_public_access_block" "origin" {
-  count                   = (local.create_s3_origin_bucket || local.override_origin_bucket_policy) && var.block_origin_public_access_enabled ? 1 : 0
+  count                   = local.create_s3_origin_bucket || local.override_origin_bucket_policy ? 1 : 0
   bucket                  = local.bucket
-  block_public_acls       = true
-  block_public_policy     = true
-  ignore_public_acls      = true
-  restrict_public_buckets = true
-
-  # Don't modify this bucket in two ways at the same time, S3 API will complain.
-  depends_on = [aws_s3_bucket_policy.default]
+  block_public_acls       = local.block_origin_public_access
+  block_public_policy     = local.block_origin_public_access
+  ignore_public_acls      = local.block_origin_public_access
+  restrict_public_buckets = local.block_origin_public_access
```

Comment:
Block unless it is a static website and we have not explicitly set the `block_origin_public_access_enabled` variable